org-roam.el

```diff
@@ -374,6 +374,26 @@ current buffer is used."
 (defun org-roam--extract-ref ()
   "Extract the ref from current buffer."
   (cdr (assoc "ROAM_KEY" (org-roam--extract-global-props '("ROAM_KEY")))))
+
+(defun org-roam--ref-type (ref)
+  "Determine the type of the REF from the prefix."
+  (let* ((cite-prefix (org-roam--cite-prefix ref))
+         (is-website (seq-some
+                      (lambda (prefix) (s-prefix? prefix ref))
+                      '("http" "https")))
+         (type (cond (cite-prefix "cite")
+                     (is-website "website")
+                     (t "roam"))))
+    type))
+
+(defun org-roam--cite-prefix (ref)
+  "Return citation prefix of REF if belongs to `org-ref-cite-types`."
```

Comment:
```suggestion
  "Return the citation prefix of REF (e.g. \"cite:\", \"parencite:\" etc.) if it has a prefix of
  one of the `org-ref-cite-types`, return `nil` otherwise. This can be used to determine if a ref
  should be a \"cite\" type"
```
A little bit more verbose for the docstring